File: general/src/main/java/com/workoutnow/general/service/util/UserUtil.java
package com.workoutnow.general.service.util;

import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.oauth2.jwt.Jwt;

public class UserUtil {
    public static String getCurrentUserId(){
        Jwt jwt = (Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        return jwt.getClaims().get("sub").toString();
    }
}

File: analytics/src/main/kotlin/com/workoutnow/analytics/service/util/UserUtil.kt
package com.workoutnow.analytics.service.util

import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.oauth2.jwt.Jwt

object UserUtil {

    var currentUserId: String = ""
        get() {
            if(field == ""){
                val jwt = SecurityContextHolder.getContext().authentication.principal as Jwt
                return jwt.claims["sub"].toString()
            }
            return field
        }

}